vm/systemSmartContracts/governance.go

```diff
@@ -385,972 +394,573 @@ func (g *governanceContract) delegateVote(args *vmcommon.ContractCallInput) vmco
-func (g *governanceContract) computeValidatorVotingPower(validatorAddress []byte) (*big.Int, error) {
-	totalStake, err := g.getTotalStake(validatorAddress)
-	if err != nil {
-		return nil, fmt.Errorf("could not return total stake for the provided address, thus cannot compute voting power")
-	}
-
-	votingPower, err := g.computeVotingPower(totalStake)
+func (g *governanceContract) saveConfig(cfg *GovernanceConfigV2) error {
+	marshaledData, err := g.marshalizer.Marshal(cfg)
 	if err != nil {
-		return nil, fmt.Errorf("could not return total stake for the provided address, thus cannot compute voting power")
+		return err
 	}
 
-	return votingPower, nil
+	g.eei.SetStorage([]byte(governanceConfigKey), marshaledData)
+	return nil
 }
 
-// function iterates over all delegation contracts and verifies balances of the given account and makes a sum of it
-//TODO: benchmark this, the other solution is to receive in the arguments which delegation contracts should be checked
-// and consume gas for each delegation contract to be checked
-func (g *governanceContract) computeVotingPowerFromTotalStake(address []byte) (*big.Int, error) {
-	totalStake, err := g.getTotalStake(address)
-	if err != nil && err != vm.ErrEmptyStorage {
-		return nil, fmt.Errorf("could not return total stake for the provided address, thus cannot compute voting power")
-	}
-	if totalStake == nil {
-		totalStake = big.NewInt(0)
-	}
-
-	dContractList, err := getDelegationContractList(g.eei, g.marshalizer, g.delegationMgrSCAddress)
-	if err != nil {
-		return nil, err
-	}
-
-	var activeDelegated *big.Int
-	for _, contract := range dContractList.Addresses {
-		activeDelegated, err = g.getActiveFundForDelegator(contract, address)
-		if err != nil {
-			return nil, err
-		}
-
-		totalStake.Add(totalStake, activeDelegated)
-	}
-
-	votingPower, err := g.computeVotingPower(totalStake)
+// saveGeneralProposal saves a proposal into the storage
+func (g *governanceContract) saveGeneralProposal(reference []byte, generalProposal *GeneralProposal) error {
+	marshaledData, err := g.marshalizer.Marshal(generalProposal)
 	if err != nil {
-		return nil, err
+		return err
 	}
+	key := append([]byte(proposalPrefix), reference...)
+	g.eei.SetStorage(key, marshaledData)
 
-	return votingPower, nil
+	return nil
 }
 
-func (g *governanceContract) getActiveFundForDelegator(delegationAddress []byte, address []byte) (*big.Int, error) {
-	dData := &DelegatorData{
-		UnClaimedRewards:      big.NewInt(0),
-		TotalCumulatedRewards: big.NewInt(0),
-	}
-	marshaledData := g.eei.GetStorageFromAddress(delegationAddress, address)
-	if len(marshaledData) == 0 {
-		return big.NewInt(0), nil
-	}
-
-	err := g.marshalizer.Unmarshal(dData, marshaledData)
+// getValidProposal returns a proposal from storage if it exists, or it is still valid/in-progress
+func (g *governanceContract) getValidProposal(nonce *big.Int) (*GeneralProposal, error) {
+	commitHash := g.eei.GetStorage(nonce.Bytes())
```

Comment:
suggestion to add a prefix here too? Just to eliminate the possible key conflicts?